test/e2e/apps/replica_set.go

```diff
@@ -252,7 +251,7 @@ func testReplicaSetConditionCheck(f *framework.Framework) {
 		conditions = rs.Status.Conditions
 
 		cond := replicaset.GetCondition(rs.Status, appsv1.ReplicaSetReplicaFailure)
-		return cond == nil, nil
+		return (cond == nil || cond.Status == v1.ConditionFalse), nil
```

Comment:
I think the condition should be always set to false in this case and replica shouldn't be removing it. Missing condition is Unknown, not false https://github.com/kubernetes/community/blame/b9312b7df521cb9caadb994ec5bb30d13cf114aa/contributors/devel/sig-architecture/api-conventions.md#L358

Reply:
Thanks for the feedback @tnozicka 

I'm not sure what you're recommending here. Are you saying that the condition should be set to False, and then Unknown if missing, or was that a separate statement?

Reply:
hm, I am not sure what value is this case adding - RS is likely wrong by removing the condition instead of setting it to false. If the controller always removes the condition instead of setting it to false, why are you changing the tests to check for false which I guess never gets hit? 
I'd understand a PR that would fix the RS controller and changed this line to reflect that but why is it changed on its own?